### PR TITLE
PixelClassificationGui: Fixes recursive FreezePredictions dirty event and cleans up code

### PR DESCRIPTION
**Pull request by Tomaz-Vieira**
## Done in this PR:

- Removes recursive handling of FreezePredictions dirty event - the handler itself was also setting the value of the slot, firing another event;
- Moves code that deals with visual configuration of the "live update" button into the proper .ui file;
- Cleans up the code a bit.

### CODE REVIEW

File: ilastik/applets/pixelClassification/pixelClassificationGui.py [hunk 1]
```diff
@@ -718,46 +702,42 @@ def setPredLayerName(n, predictLayer_=predictLayer, initializing=False):
...
                 layers.append(predictLayer)
-        
-        self.handleLabelSelectionChange()
         return layers
 
-    def toggleInteractive(self, checked):
-        logger.debug("toggling interactive mode to '%r'" % checked)
+    def hasFeatures(self) -> bool:
+        feature_images_slot = self.topLevelOperatorView.FeatureImages
+        return feature_images_slot.ready() and feature_images_slot.meta.shape != None
```
Comment (emilmelnikov) [inline ilastik/applets/pixelClassification/pixelClassificationGui.py]:
```suggestion
        return feature_images_slot.ready() and feature_images_slot.meta.shape is not None
```

Also, this method can be a property.

File: ilastik/applets/pixelClassification/FeatureSelectionDialog.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ class FeatureSelectionDialog(QtWidgets.QDialog):
     def __init__(
             self,
             current_opFeatureSelection,
-            current_opPixelClassification,
+            current_opPixelClassificationApplet,
```
[the remark this answers is not in the document]
Comment (emilmelnikov) [reply ilastik/applets/pixelClassification/FeatureSelectionDialog.py]: What is the meaning of the `current_` prefixes? Maybe just leave them out?

File: ilastik/applets/pixelClassification/pixelClassificationGui.py [hunk 3]
```diff
@@ -718,46 +702,42 @@ def setPredLayerName(n, predictLayer_=predictLayer, initializing=False):
...
+            self.labelingDrawerUi.AddLabelButton.setEnabled( False )
 
-        # Auto-set the "show predictions" state according to what the user just clicked.
-        if checked:
             self._viewerControlUi.checkShowPredictions.setChecked( True )
             self.handleShowPredictionsClicked()
+        else:
+            self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
+            if not self.hasFeatures():
+                mexBox=QMessageBox()
```
Comment (emilmelnikov) [inline ilastik/applets/pixelClassification/pixelClassificationGui.py]:
```suggestion
                mexBox = QMessageBox()
```

Or maybe even something like `QMessageBox.warning(self, "Features", "No features selected")`.

Also, maybe remove nested condition by duplicating the `suggestFeaturesButton.setEnabled(False)` call? Seems a bit easier to follow.

```python
if checked and self.hasFeatures():
    ...
elif not self.hasFeatures():
    self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
    QMessageBox.warning(self, "Features", "No features selected")
    return
else:
    self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
    ....
```

File: ilastik/applets/pixelClassification/pixelClassificationGui.py [hunk 4]
```diff
@@ -718,46 +702,42 @@ def setPredLayerName(n, predictLayer_=predictLayer, initializing=False):
...
+                mexBox.setText("There are no features selected ")
+                mexBox.exec_()
+                return
+
+            num_label_classes = self._labelControlUi.labelListModel.rowCount()
+            self.labelingDrawerUi.labelListView.allowDelete = ( num_label_classes > self.minLabelNumber )
+            self.labelingDrawerUi.AddLabelButton.setEnabled( ( num_label_classes < self.maxLabelNumber ) )
```
Comment (emilmelnikov) [inline ilastik/applets/pixelClassification/pixelClassificationGui.py]: Stray parens?

```suggestion
            self.labelingDrawerUi.AddLabelButton.setEnabled(num_label_classes < self.maxLabelNumber)
```

File: ilastik/applets/pixelClassification/pixelClassificationGui.py (hunk 3 above)
Comment (Tomaz-Vieira) [reply ilastik/applets/pixelClassification/pixelClassificationGui.py]: Yeah, that part did feel weird. Turns out it was needlessly convoluted. I've streamlined the code a bit more

File: ilastik/applets/pixelClassification/pixelClassificationGui.py [hunk 5]
```diff
@@ -718,46 +702,42 @@ def setPredLayerName(n, predictLayer_=predictLayer, initializing=False):
...
-            if not self.topLevelOperatorView.FeatureImages.ready() \
-            or self.topLevelOperatorView.FeatureImages.meta.shape==None:
-                self.labelingDrawerUi.liveUpdateButton.setChecked(False)
-                self.labelingDrawerUi.liveUpdateButton.setIcon(QIcon(ilastikIcons.Play))
-                self.labelingDrawerUi.suggestFeaturesButton.setEnabled(False)
-                mexBox=QMessageBox()
-                mexBox.setText("There are no features selected ")
-                mexBox.exec_()
-                return
+    def getLiveUpdateFlag(self):
```
[the remark this answers is not in the document]
Comment (Tomaz-Vieira) [reply ilastik/applets/pixelClassification/pixelClassificationGui.py]: Changed method to `isLiveUpdateEnabled`. I think changing it into a property might make it harder to convey that it represents a boolean (and also that it it's causing a bunch of side-effects, that is, it's allowing predicitons to carry on)

File: ilastik/applets/pixelClassification/pixelClassificationGui.py (hunk 4 above)
Comment (Tomaz-Vieira) [reply ilastik/applets/pixelClassification/pixelClassificationGui.py]: Yup. Legacy code. Done =)

**Pull request merged**